pkg/utils/runnable_server.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// RunnableServer is a small wrapper around http.Server so that it matches the
+// manager.Runnable interface.
+type RunnableServer struct {
+	// Server is the http.Server to wrap.
+	*http.Server
+
+	// ServeFunc is the function used to start the http.Server. If nil,
+	// ListenAndServe() will be used.
+	ServeFunc func() error
```

Comment:
This seems to only be used by test code, is it needed?

Reply:
The case for a user needing this is a non-default Listener (unix sockets, udp, etc) or TLS. TLS seems likely to come up eventually, but if this seems ugly to you I'd be fine with making it internal.